File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x: x[1])
        prev_end = float('-inf')
        count = 0

        for interval in intervals:
            if interval[0] < prev_end:
                count += 1
            else:
                prev_end = interval[1]

        return count

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        int min_erase = 0;
        int prev_end = INT_MIN;

        sort(intervals.begin(), intervals.end(), [](const vector<int>& a, const vector<int>& b) {
            return a[1] < b[1];
        });

        for (const auto interval : intervals) {
            if (interval[0] >= prev_end) {
                prev_end = interval[1];
            }
            else {
                min_erase++;
            }
        }

        return min_erase;
    }
};